File: admin-web/src/service/login.service.ts
import { axiosInstance } from "@/lib/axios"

interface LoginBody {
	username: string
	password: string
}

interface LoginResponse {
	success: boolean
	token: string
}

class LoginServiceDefinition {
	version: string

	constructor(version: string) {
		this.version = version
	}

	setVersion(version: string) {
		this.version = version
	}

	async login(body: LoginBody): Promise<LoginResponse> {
		try {
			const res = await axiosInstance.post(`/${this.version}/auth`, body)
			// return res.data
			return { success: true, token: res.data }
		} catch (error) {
			console.error(error)
			return { success: false, token: "" }
		}
	}

	async logout(): Promise<Boolean> {
		try {
			window.localStorage.removeItem("token")
			return true
		} catch (error) {
			console.error(error)
			return false
		}
	}
}

const LoginService = new LoginServiceDefinition("v1")
export default LoginService

File: mobile_app/lib/services/login_service.dart
import 'package:mobile_app/config/dio.dart';

class LoginService {
  static Future<String?> login(username, password) async {
    try {
      final ans = await dioInstance.post("/v1/auth", data: {
        'username': username,
        'password': password,
      });
      return ans.data;
    } catch (e) {
      print(e);
      return null;
    }
  }
}
